### PR TITLE
Enforce spaces inside parenthesis

### LINKED ISSUE #3855: Enforce spaces inside parenthesis
I am trying to get PHP-CS-Fixer to enforce spaces inside parenthesis. I have found the fixer "no_spaces_inside_parenthesis" which, when enabled, will strip spaces inside parenthesis. But disabling this fixer does not enforce spaces inside parenthesis. This behaviour seems correct and expected to me, but I am not seeing a way to achieve my goal in any other way. Do you know if it is possible to make PHP-CS-Fixer enforce spaces inside parenthesis?

When reporting an issue (bug) please provide the following information:

#### The PHP version you are using (`$ php -v`):
=>
`PHP 7.1.15 (cli) (built: Mar  1 2018 01:51:52) ( ZTS MSVC14 (Visual C++ 2015) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans`

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
`PHP CS Fixer 2.12.1 Long Journey by Fabien Potencier and Dariusz Ruminski`

#### The command you use to run PHP CS Fixer:
`php-cs-fixer fix  .\website --verbose --config=config.php_cs`

#### The configuration file you are using, if any:
```php
<?php

$finder = PhpCsFixer\Finder::create()
    ->exclude('vendor')
    ->in(__DIR__)
;

return PhpCsFixer\Config::create()
    ->setIndent("\t")
    ->setLineEnding("\r\n")
    ->setRules([
        '@Symfony' => true,
		'array_syntax' => ['syntax' => 'long'],
		'blank_line_after_namespace' => true,
		'blank_line_after_opening_tag' => true,
		'blank_line_before_return' => false,
		'blank_line_before_statement' => false,
		'function_declaration' => ['closure_function_spacing' => 'one'],
		'braces' => [
			'allow_single_line_closure' => true,
			'position_after_anonymous_constructs' => 'same',
			'position_after_control_structures' => 'same',
			'position_after_functions_and_oop_constructs' => 'same'
		],
		'full_opening_tag' => true,
		'no_closing_tag' => true,
		'no_trailing_whitespace_in_comment' => false,
		'whitespace_after_comma_in_array' => true,
		'no_spaces_inside_parenthesis' => false,
		'phpdoc_no_package' => false,
		'method_argument_space' => true,
		'no_spaces_around_offset' => ['outside'],
		'concat_space' => [ 'spacing' => 'one' ],
		'no_spaces_after_function_name' => false,
		'general_phpdoc_annotation_remove' => false,
		'include' => true,
		'array_indentation' => true,
		'trim_array_spaces' => false,
		'cast_spaces' => ['space' => 'single']
    ])
    ->setFinder($finder)
;

```

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):
 * before running PHP CS Fixer (no changes):
```php
explode($a, $b);
```

 * with unexpected changes applied when running PHP CS Fixer:
```php
explode($a, $b);
```

 * with the changes you expected instead:
```php
explode( $a, $b );
```

### PR DESCRIPTION
**Pull request by tareq1988**
Yes, the WordPress community is following the [space usages](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage) inside the parenthesis. The issue #3855 also mentions this.

This pull request fixes #3855

### CODE REVIEW

**Comment by keradus** [comment]
hey @tareq1988 ,
as you already noticed on gitter, this new rule is in conflict with `no_spaces_inside_parenthesis`.
Imagine what will be the outcome of someone enabling 2 rules at same time. we do not want to have such conflicting fixers.

Instead, there should be a single fixer with a configuration - should it add or remove spaces.

File: src/Fixer/Whitespace/SpaceInsideParenthesisFixer.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Fixer for rules defined in PSR2 generally (¶1 and ¶6).
```
Comment (keradus) [inline src/Fixer/Whitespace/SpaceInsideParenthesisFixer.php]: I don't think this is part of PSR2

File: tests/Fixer/Whitespace/SpaceInsideParenthesisFixerTest.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    // code...
+}
+',
+            ],
+            [
+                '<?php
+explode( $a, $b );
+',
+                '<?php
+explode($a, $b);
```
Comment (keradus) [inline tests/Fixer/Whitespace/SpaceInsideParenthesisFixerTest.php]: what about equations ? eg `(2+3) * 4`

**Comment by tareq1988** [comment]
Yes, I've made this opposite of `no_spaces_inside_parenthesis`. But as the name suggests, there should be no space inside the parenthesis. So I guess adding option to add spaces inside parenthesis won't respect the name itself 😃

File: src/Fixer/Whitespace/SpaceInsideParenthesisFixer.php (hunk 1 above)
Comment (tareq1988) [reply src/Fixer/Whitespace/SpaceInsideParenthesisFixer.php]: That was a copy/paste issue.

File: tests/Fixer/Whitespace/SpaceInsideParenthesisFixerTest.php (hunk 2 above)
Comment (tareq1988) [reply tests/Fixer/Whitespace/SpaceInsideParenthesisFixerTest.php]: Added tests

**Comment by kubawerlos** [comment]
> there should be a single fixer

@keradus meant creating new fixer, with a name something like `spaces_inside_parenthesis_style` which would have config and deprecate `no_spaces_inside_parenthesis` in favour of it.

Similar change to `increment_style` [replacing](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3065/files) `pre_increment`.

**Comment by tareq1988** [comment]
Yes, that certainly makes sense.

**Pull request closed without merging**